### PR TITLE
ROR Recipe app group project module 5

### PR DESCRIPTION
**Pull request by juaryimami**
### In this PR:-
---
- We follow the layout of the wireframes provided. And personalized the rest of the design including colors, typographies, spacings, etc.
- Login page and registration page:
- - Its  built with Devise.
- Food list is added:
- -  displays a list of food added by the logged-in user as in the `wireframe` (display also quantity of a given food).
It leads to a form that allows users to add new food.
- Recipes list:
- - A list of recipes created by the logged-in user as in the `wireframe`.
- - leads to recipe details.
- - The user is the owner of the recipe, and allows the user to delete it.
- Public recipe list:
- - It displays a list of all public recipes ordered by newest as in the wireframe.
- -  Leads to recipe details.
- Recipe details:
- -  displays a toggle button that allows for a recipe to be made public or private.
- - If the recipe is public or the user is the owner of the recipe, our page display the recipe details as in the wireframe.
- -If the user is the owner of the recipe, our page lead to the form that allows the user to add new food.
- General shopping list view is added:
- -  shows the list of food that is missing for all recipes of the logged-in user (compare the list of food for all recipes with the general food list of that user).
- -  counts the total food items and total price of the missing food.
- There are no N+1 queries happening.
- We created a navigation menu that allows users to open all of the pages we created.
-  unit and integration tests are performed.
---

### CODE REVIEW

**Review by Davidosky007 — CHANGES_REQUESTED** [review]
Hi @juaryimami @r-ahmed2022 @habasm 👋, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
* All tests passed 👍 
* Rails naming conventions understood 👍 
* No linter errors 👍 

## Required Changes ♻️

_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @davidsoky007 in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/controllers/foods_controller.rb [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+  def index
+    @foods = current_user.foods.all
+  end
+
+  def new
+    @food = Food.new
+  end
+
+  def create
+    @food = Food.create(food_params.merge(user_id: current_user.id))
```
[the remark this answers is not in the document]
Comment (habasm) [reply app/controllers/foods_controller.rb:11]: @Davidosky007 Thanks for comment but we have added all
- Shopping List
image
- Add Ingredient
image
- Recipe details
image
- Allow as to create new food interface
image

### And then we added different interface you might not consider. Please add a detail comment as much as possible.

[the remark this answers is not in the document]
Comment (juaryimami) [reply app/controllers/foods_controller.rb:11]: Dear reviewer as the start of this project we were team of three but unfortunately @r-ahmed2022 left from the team and that is the reason.

**Review by Davidosky007 — APPROVED** [review]
Hi @juaryimami @habasm  👋, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
* All tests passed 👍
* Rails naming conventions understood 👍
* No linter errors 👍
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**